server/src/main/java/org/cloudfoundry/identity/uaa/zone/IdentityZoneEndpoints.java

```diff
@@ -395,6 +404,54 @@ private ClientDetails removeSecret(ClientDetails createdClient) {
         }
     }
 
+    @RequestMapping(method = POST, value = "{identityZoneId}/key-provider-config")
+    public ResponseEntity<KeyProviderConfig> createOrUpdateKeyProviderConfig(@RequestBody KeyProviderConfig body, @PathVariable String identityZoneId) throws KeyProviderValidator.KeyProviderValidatorException {
+        validateZoneId(identityZoneId);
+        body.setIdentityZoneId(identityZoneId);
+        keyProviderValidator.validate(body);
+        KeyProviderConfig keyProvider = keyProviderProvisioning.createOrUpdate(body);
+        return new ResponseEntity<>(keyProvider, CREATED);
+    }
+
+    @RequestMapping(method = GET, value="{identityZoneId}/key-provider-config/{keyProviderId}")
```

Comment:
Getting by id AND by active goes wierdly with the current createOrUpdate idea. We should pick a side: Either allow multiple key provider configs and have an active a la MFA endpoints, or stick with a single key provider config per zone and allow createOrUpdate. 

We think single key provider config per zone is a simpler solution.

Reply:
The idea is being RESTful. So, no `createOrUpdate`. We support only `create` and `delete`. We will support only one `keyProviderConfig` per zone for now but this way it will be easy to expand if needed. Update is indirectly possible by doing a delete followed by a create. This may be ok because it is probably not heavily used. We will rename the methods based on `create` only assumption